In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)

In [2]:
orders = pd.read_csv('../Datos_Brutos/global_superstore_2016_orders.csv') # Cargamos el archivo csv.

orders.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region,Market,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority
0,40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High
1,26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical
2,25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium
3,13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium
4,47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical


In [3]:
orders.shape # Comprobamos numero de filas y columnas.

(51290, 24)

In [4]:
orders.info() # Hacemos una vista previa de los datos.

<class 'pandas.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 24 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Row ID          51290 non-null  int64  
 1   Order ID        51290 non-null  str    
 2   Order Date      51290 non-null  str    
 3   Ship Date       51290 non-null  str    
 4   Ship Mode       51290 non-null  str    
 5   Customer ID     51290 non-null  str    
 6   Customer Name   51290 non-null  str    
 7   Segment         51290 non-null  str    
 8   Postal Code     9994 non-null   float64
 9   City            51290 non-null  str    
 10  State           51290 non-null  str    
 11  Country         51290 non-null  str    
 12  Region          51290 non-null  str    
 13  Market          51290 non-null  str    
 14  Product ID      51290 non-null  str    
 15  Category        51290 non-null  str    
 16  Sub-Category    51290 non-null  str    
 17  Product Name    51290 non-null  str    
 1

In [5]:
orders.describe().T

,count,mean,std,min,25%,50%,75%,max
Row ID,51290.0,25645.500000,14806.291990,1.000,12823.25,25645.50,38467.75,51290.00
Postal Code,9994.0,55190.379428,32063.693350,1040.000,23223.00,56430.50,90008.00,99301.00
Quantity,51290.0,3.476545,2.278766,1.000,2.00,3.00,5.00,14.00
Discount,51290.0,0.142908,0.212280,0.000,0.00,0.00,0.20,0.85
Shipping Cost,51290.0,26.478567,57.251373,1.002,2.61,7.79,24.45,933.57


In [6]:
orders.describe(include=str).T

,count,unique,top,freq
Order ID,51290,25728,CA-2015-SV20365140-42268,14
Order Date,51290,1430,6/18/2015,135
Ship Date,51290,1464,11/22/2015,130
Ship Mode,51290,4,Standard Class,30775
Customer ID,51290,17415,SV-203651406,26
Customer Name,51290,796,Muhammed Yedwab,108
Segment,51290,3,Consumer,26518
City,51290,3650,New York City,915
State,51290,1102,California,2001
Country,51290,165,United States,9994


In [7]:
peoples = pd.read_excel('../Datos_Brutos/global_superstore_2016_peoples.xlsx') # Cargamos el segundo archivo.

peoples.head()

,Person,Region
0,Marilène Rousseau,Caribbean
1,Andile Ihejirika,Central Africa
2,Nicodemo Bautista,Central America
3,Cansu Peynirci,Central Asia
4,Lon Bonher,Central US


In [8]:
peoples.info # Hacemos una vista previa

<bound method DataFrame.info of                   Person             Region
0      Marilène Rousseau          Caribbean
1       Andile Ihejirika     Central Africa
2      Nicodemo Bautista    Central America
3         Cansu Peynirci       Central Asia
4             Lon Bonher         Central US
5           Wasswa Ahmed     Eastern Africa
6          Hadia Bousaid       Eastern Asia
7         Lynne Marchand     Eastern Canada
8          Oxana Lagunov     Eastern Europe
9          Dolores Davis         Eastern US
10      Lindiwe Afolayan       North Africa
11          Miina Nylund    Northern Europe
12           Kauri Anaru            Oceania
13       Vasco Magalhães      South America
14      Preecha Metharom  Southeastern Asia
15          Nora Cuijper    Southern Africa
16  Chandrakant Chaudhri      Southern Asia
17           Gavino Bove    Southern Europe
18       Flannery Newton        Southern US
19        Katlego Akosua     Western Africa
20             Kaoru Xun       Western Asia


In [9]:
dataset = orders.merge(peoples, on='Region', how='left') # Unimos los dos dataframes por Region. Manteniendo todos los pedidos.

dataset.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region,Market,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority,Person
0,40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher
1,26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru
2,25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru
3,13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff
4,47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua


In [10]:
dataset.shape # Comprobamos que se han  unido bien.

(51290, 25)

In [11]:
dataset['Person'].value_counts()

Person
Gilbert Wolff           5883
Nicodemo Bautista       5616
Kauri Anaru             3487
Derrick Snyders         3203
Preecha Metharom        3129
Vasco Magalhães         2988
Dolores Davis           2848
Chandrakant Chaudhri    2655
Kaoru Xun               2440
Hadia Bousaid           2374
Lon Bonher              2323
Miina Nylund            2204
Gavino Bove             2113
Marilène Rousseau       1690
Flannery Newton         1620
Oxana Lagunov           1529
Katlego Akosua          1460
Lindiwe Afolayan        1278
Wasswa Ahmed             728
Andile Ihejirika         643
Nora Cuijper             478
Cansu Peynirci           217
Name: count, dtype: int64

In [12]:
dataset['Person'].isnull().sum() # Comprobamos nulos.

np.int64(384)

In [13]:
dataset[dataset['Person'].isnull()]['Region'].unique() # Comprobamos cuales son los nulos.

<StringArray>
['Canada']
Length: 1, dtype: str

In [14]:
dataset.loc[dataset['Region'] == 'Canada', 'Person'] = 'Gerente Canada' # Completamos el dataset con 'Gerente Canada' al no tener el nombre.

In [15]:
dataset['Person'].isnull().sum()

np.int64(0)

In [16]:
dataset.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region,Market,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority,Person
0,40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher
1,26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru
2,25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru
3,13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff
4,47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua


In [17]:
returns = pd.read_excel('../Datos_Brutos/global_superstore_2016_returns.xlsx') # Cargamos el archivo devoluciones.

returns.head()

,Returned,Order ID,Region
0,Yes,CA-2012-SA20830140-41210,Central US
1,Yes,IN-2012-PB19210127-41259,Eastern Asia
2,Yes,CA-2012-SC20095140-41174,Central US
3,Yes,IN-2015-JH158207-42140,Oceania
4,Yes,IN-2014-LC168857-41747,Oceania


In [18]:
returns.shape # Vista previa

(1079, 3)

In [19]:
returns.info()

<class 'pandas.DataFrame'>
RangeIndex: 1079 entries, 0 to 1078
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   Returned  1079 non-null   str  
 1   Order ID  1079 non-null   str  
 2   Region    1079 non-null   str  
dtypes: str(3)
memory usage: 25.4 KB


In [20]:
dataset = dataset.merge(returns, on='Order ID', how='left') # Y unimos.

dataset.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region_x,Market,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority,Person,Returned,Region_y
0,40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher,NaN,NaN
1,26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru,NaN,NaN
2,25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru,NaN,NaN
3,13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff,NaN,NaN
4,47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua,NaN,NaN


In [21]:
dataset = dataset.drop(columns=['Region_y']) # Quitamos la columna region repetida.

dataset.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Postal Code,City,State,Country,Region_x,Market,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Shipping Cost,Order Priority,Person,Returned
0,40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher,NaN
1,26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru,NaN
2,25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru,NaN
3,13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff,NaN
4,47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua,NaN


In [22]:
dataset.columns # Vemos que columnas tenemos.

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Postal Code', 'City',
       'State', 'Country', 'Region_x', 'Market', 'Product ID', 'Category',
       'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount',
       'Profit', 'Shipping Cost', 'Order Priority', 'Person', 'Returned'],
      dtype='str')

In [23]:
dataset['Market'].value_counts() # Vemos la columna Market para ver que es.

Market
Asia Pacific    14302
Europe          11729
USCA            10378
LATAM           10294
Africa           4587
Name: count, dtype: int64

In [24]:
# Creamos un diccionario para traducir los encabezados de las columnas.

traduccion = {
    'Row ID': 'ID_Venta',
    'Order ID': 'ID_Pedido',
    'Order Date': 'Fecha_Pedido',
    'Ship Date': 'Fecha_Envio',
    'Ship Mode': 'Metodo_Envio',
    'Customer ID': 'ID_Cliente',
    'Customer Name': 'Nombre_Cliente',
    'Segment': 'Perfil_Cliente',
    'Postal Code': 'Codigo_Postal',
    'City': 'Ciudad',
    'State': 'Estado',
    'Country': 'Pais',
    'Region_x': 'Region',
    'Market': 'Macro_Zona',
    'Product ID': 'ID_Producto',
    'Category': 'Categoria',
    'Sub-Category': 'Subcategoria',
    'Product Name': 'Nombre_Producto',
    'Sales': 'Ventas',
    'Quantity': 'Cantidad',
    'Discount': 'Descuento',
    'Profit': 'Beneficio',
    'Shipping Cost': 'Gastos_Envio',
    'Order Priority': 'Prioridad_Pedido',
    'Person': 'Gerente',
    'Returned': 'Devuelto'
}

In [25]:
dataset = dataset.rename(columns=traduccion)

dataset.head()

,ID_Venta,ID_Pedido,Fecha_Pedido,Fecha_Envio,Metodo_Envio,ID_Cliente,Nombre_Cliente,Perfil_Cliente,Codigo_Postal,Ciudad,Estado,Pais,Region,Macro_Zona,ID_Producto,Categoria,Subcategoria,Nombre_Producto,Ventas,Cantidad,Descuento,Beneficio,Gastos_Envio,Prioridad_Pedido,Gerente,Devuelto
0,40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher,NaN
1,26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru,NaN
2,25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru,NaN
3,13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff,NaN
4,47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua,NaN


In [26]:
dataset['ID_Venta'].value_counts() 

ID_Venta
40098    1
26341    1
25330    1
13524    1
47221    1
        ..
29002    1
34337    1
31315    1
9596     1
6147     1
Name: count, Length: 51290, dtype: int64

In [27]:
dataset['ID_Venta'].describe()

count    51290.00000
mean     25645.50000
std      14806.29199
min          1.00000
25%      12823.25000
50%      25645.50000
75%      38467.75000
max      51290.00000
Name: ID_Venta, dtype: float64

In [28]:
dataset = dataset.set_index('ID_Venta') # Ponemos el ID_Venta como indice

dataset.head()

,ID_Pedido,Fecha_Pedido,Fecha_Envio,Metodo_Envio,ID_Cliente,Nombre_Cliente,Perfil_Cliente,Codigo_Postal,Ciudad,Estado,Pais,Region,Macro_Zona,ID_Producto,Categoria,Subcategoria,Nombre_Producto,Ventas,Cantidad,Descuento,Beneficio,Gastos_Envio,Prioridad_Pedido,Gerente,Devuelto
ID_Venta,,,,,,,,,,,,,,,,,,,,,,,,,
40098,CA-2014-AB10015140-41954,11/11/2014,11/13/2014,First Class,AB-100151402,Aaron Bergman,Consumer,73120.0,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher,NaN
26341,IN-2014-JR162107-41675,2/5/2014,2/7/2014,Second Class,JR-162107,Justin Ritter,Corporate,NaN,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru,NaN
25330,IN-2014-CR127307-41929,10/17/2014,10/18/2014,First Class,CR-127307,Craig Reiter,Consumer,NaN,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru,NaN
13524,ES-2014-KM1637548-41667,1/28/2014,1/30/2014,First Class,KM-1637548,Katherine Murray,Home Office,NaN,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff,NaN
47221,SG-2014-RH9495111-41948,11/5/2014,11/6/2014,Same Day,RH-9495111,Rick Hansen,Consumer,NaN,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua,NaN


In [29]:
dataset.info()

<class 'pandas.DataFrame'>
Index: 51290 entries, 40098 to 6147
Data columns (total 25 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   ID_Pedido         51290 non-null  str    
 1   Fecha_Pedido      51290 non-null  str    
 2   Fecha_Envio       51290 non-null  str    
 3   Metodo_Envio      51290 non-null  str    
 4   ID_Cliente        51290 non-null  str    
 5   Nombre_Cliente    51290 non-null  str    
 6   Perfil_Cliente    51290 non-null  str    
 7   Codigo_Postal     9994 non-null   float64
 8   Ciudad            51290 non-null  str    
 9   Estado            51290 non-null  str    
 10  Pais              51290 non-null  str    
 11  Region            51290 non-null  str    
 12  Macro_Zona        51290 non-null  str    
 13  ID_Producto       51290 non-null  str    
 14  Categoria         51290 non-null  str    
 15  Subcategoria      51290 non-null  str    
 16  Nombre_Producto   51290 non-null  str    
 17  Ventas

In [30]:
dataset['ID_Pedido'].value_counts()

ID_Pedido
CA-2015-SV20365140-42268    14
IN-2014-MH1778527-41697     13
MX-2015-PO1885082-42279     13
IN-2013-TB21055113-41562    13
NI-2015-TC1098095-42033     13
                            ..
MX-2015-HG1502518-42164      1
IN-2015-DB132707-42221       1
IN-2015-KH1633058-42154      1
US-2014-ZD21925140-41765     1
MX-2013-MC1810093-41416      1
Name: count, Length: 25728, dtype: int64

Comprobamos en que tipo de dato estan las fechas, y las cambiamos de string a formato fecha.

In [31]:
dataset['Fecha_Pedido'].value_counts()

Fecha_Pedido
6/18/2015     135
11/18/2015    127
9/3/2015      126
11/20/2015    118
12/29/2015    116
             ... 
7/21/2014       1
5/27/2013       1
11/10/2014      1
2/13/2012       1
2/6/2012        1
Name: count, Length: 1430, dtype: int64

In [32]:
dataset['Fecha_Pedido'].dtype

<StringDtype(storage='python', na_value=nan)>

In [33]:
dataset['Fecha_Pedido'] = pd.to_datetime (dataset['Fecha_Pedido'])

In [34]:
dataset['Fecha_Pedido'].dtype

dtype('<M8[us]')

In [35]:
dataset['Fecha_Envio'].value_counts()

Fecha_Envio
11/22/2015    130
9/7/2015      115
12/7/2015     101
11/17/2015    101
11/29/2015    100
             ... 
1/25/2012       3
1/7/2016        3
1/3/2012        2
1/6/2012        2
1/5/2012        2
Name: count, Length: 1464, dtype: int64

In [36]:
dataset['Fecha_Envio'].dtype

<StringDtype(storage='python', na_value=nan)>

In [37]:
dataset['Fecha_Envio'] = pd.to_datetime (dataset['Fecha_Envio'])

In [38]:
dataset['Fecha_Envio'].dtype

dtype('<M8[us]')

In [39]:
dataset['Metodo_Envio'].value_counts()

Metodo_Envio
Standard Class    30775
Second Class      10309
First Class        7505
Same Day           2701
Name: count, dtype: int64

In [40]:
dataset['ID_Cliente'].value_counts()

ID_Cliente
SV-203651406    26
WB-218501404    24
AP-109151404    23
EM-1396082      21
CS-121757       20
                ..
GK-1462036       1
JR-570095        1
PM-894038        1
AS-285147        1
MC-1810093       1
Name: count, Length: 17415, dtype: int64

In [41]:
dataset['Nombre_Cliente'].value_counts()

Nombre_Cliente
Muhammed Yedwab    108
Steven Ward        106
Bill Eplett        102
Gary Hwang         102
Patrick O'Brill    102
                  ... 
David Bremer        34
Darren Budd         31
Nicole Brennan      31
Michael Oakman      29
Kai Rey              7
Name: count, Length: 796, dtype: int64

In [42]:
dataset['Perfil_Cliente'].value_counts()

Perfil_Cliente
Consumer       26518
Corporate      15429
Home Office     9343
Name: count, dtype: int64

In [43]:
dataset['Codigo_Postal'].value_counts()

Codigo_Postal
10035.0    263
10024.0    230
10009.0    229
94122.0    203
10011.0    193
          ... 
52402.0      1
61832.0      1
21740.0      1
94403.0      1
46368.0      1
Name: count, Length: 631, dtype: int64

Solo tenemos 9994 filas con codigo postal y tenemos la ciudad. En unas ventas mundiales no lo vamos a usar.

In [44]:
dataset = dataset.drop(columns= ['Codigo_Postal'])

dataset.head()

,ID_Pedido,Fecha_Pedido,Fecha_Envio,Metodo_Envio,ID_Cliente,Nombre_Cliente,Perfil_Cliente,Ciudad,Estado,Pais,Region,Macro_Zona,ID_Producto,Categoria,Subcategoria,Nombre_Producto,Ventas,Cantidad,Descuento,Beneficio,Gastos_Envio,Prioridad_Pedido,Gerente,Devuelto
ID_Venta,,,,,,,,,,,,,,,,,,,,,,,,
40098,CA-2014-AB10015140-41954,2014-11-11,2014-11-13,First Class,AB-100151402,Aaron Bergman,Consumer,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,$221.98,2,0.0,$62.15,40.77,High,Lon Bonher,NaN
26341,IN-2014-JR162107-41675,2014-02-05,2014-02-07,Second Class,JR-162107,Justin Ritter,Corporate,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black","$3,709.40",9,0.1,$-288.77,923.63,Critical,Kauri Anaru,NaN
25330,IN-2014-CR127307-41929,2014-10-17,2014-10-18,First Class,CR-127307,Craig Reiter,Consumer,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID","$5,175.17",9,0.1,$919.97,915.49,Medium,Kauri Anaru,NaN
13524,ES-2014-KM1637548-41667,2014-01-28,2014-01-30,First Class,KM-1637548,Katherine Murray,Home Office,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless","$2,892.51",5,0.1,$-96.54,910.16,Medium,Gilbert Wolff,NaN
47221,SG-2014-RH9495111-41948,2014-11-05,2014-11-06,Same Day,RH-9495111,Rick Hansen,Consumer,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed","$2,832.96",8,0.0,$311.52,903.04,Critical,Katlego Akosua,NaN


In [45]:
dataset['Ciudad'].value_counts()

Ciudad
New York City     915
Los Angeles       747
Philadelphia      537
San Francisco     510
Santo Domingo     443
                 ... 
Levin               1
Edirne              1
San Mateo           1
Portage             1
Victoria Falls      1
Name: count, Length: 3650, dtype: int64

In [46]:
dataset['Estado'].value_counts()

Estado
California            2001
England               1499
New York              1128
Texas                  985
Ile-de-France          981
                      ... 
Paramaribo               1
Kabarole                 1
Bitola                   1
Edirne                   1
Matabeleland North       1
Name: count, Length: 1102, dtype: int64

In [47]:
dataset['Pais'].value_counts()

Pais
United States    9994
Australia        2837
France           2827
Mexico           2635
Germany          2063
                 ... 
Kuwait              1
Malawi              1
Bhutan              1
Suriname            1
The Gambia          1
Name: count, Length: 165, dtype: int64

In [48]:
dataset['Region'].value_counts()

Region
Western Europe       5883
Central America      5616
Oceania              3487
Western US           3203
Southeastern Asia    3129
South America        2988
Eastern US           2848
Southern Asia        2655
Western Asia         2440
Eastern Asia         2374
Central US           2323
Northern Europe      2204
Southern Europe      2113
Caribbean            1690
Southern US          1620
Eastern Europe       1529
Western Africa       1460
North Africa         1278
Eastern Africa        728
Central Africa        643
Southern Africa       478
Canada                384
Central Asia          217
Name: count, dtype: int64

In [49]:
dataset['Macro_Zona'].value_counts()

Macro_Zona
Asia Pacific    14302
Europe          11729
USCA            10378
LATAM           10294
Africa           4587
Name: count, dtype: int64

In [50]:
dataset['Ventas'].value_counts()

Ventas
$12.96     67
$25.92     50
$19.44     43
$32.40     42
$15.55     41
           ..
$4.05       1
$2.26       1
$0.44       1
$213.48     1
$259.96     1
Name: count, Length: 20049, dtype: int64

In [51]:
dataset['Beneficio'].value_counts()

Beneficio
$0.00      672
$4.32       73
$7.92       72
$3.96       72
$2.88       71
          ... 
$47.51       1
$-23.33      1
$-13.26      1
$124.78      1
$-49.57      1
Name: count, Length: 16577, dtype: int64

Sabemos que 'Ventas' y 'Beneficio' estan en str hay que cambiardo a float

In [52]:
dataset['Ventas'].dtype

<StringDtype(storage='python', na_value=nan)>

In [53]:
dataset['Beneficio'].dtype

<StringDtype(storage='python', na_value=nan)>

Creamos una funcion para quitar el simbolo $ y las comas.

In [54]:
def limpiar_numeros (dato):
    dato = dato.replace('$', '').replace(',', '')
    return float (dato)


In [55]:
dataset['Ventas'] = dataset['Ventas'].apply(limpiar_numeros)

dataset['Beneficio'] = dataset['Beneficio'].apply(limpiar_numeros)

In [56]:
dataset['Ventas'].dtype

dtype('float64')

In [57]:
dataset['Beneficio'].dtype

dtype('float64')

In [58]:
dataset.head()

,ID_Pedido,Fecha_Pedido,Fecha_Envio,Metodo_Envio,ID_Cliente,Nombre_Cliente,Perfil_Cliente,Ciudad,Estado,Pais,Region,Macro_Zona,ID_Producto,Categoria,Subcategoria,Nombre_Producto,Ventas,Cantidad,Descuento,Beneficio,Gastos_Envio,Prioridad_Pedido,Gerente,Devuelto
ID_Venta,,,,,,,,,,,,,,,,,,,,,,,,
40098,CA-2014-AB10015140-41954,2014-11-11,2014-11-13,First Class,AB-100151402,Aaron Bergman,Consumer,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,221.98,2,0.0,62.15,40.77,High,Lon Bonher,NaN
26341,IN-2014-JR162107-41675,2014-02-05,2014-02-07,Second Class,JR-162107,Justin Ritter,Corporate,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black",3709.40,9,0.1,-288.77,923.63,Critical,Kauri Anaru,NaN
25330,IN-2014-CR127307-41929,2014-10-17,2014-10-18,First Class,CR-127307,Craig Reiter,Consumer,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID",5175.17,9,0.1,919.97,915.49,Medium,Kauri Anaru,NaN
13524,ES-2014-KM1637548-41667,2014-01-28,2014-01-30,First Class,KM-1637548,Katherine Murray,Home Office,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless",2892.51,5,0.1,-96.54,910.16,Medium,Gilbert Wolff,NaN
47221,SG-2014-RH9495111-41948,2014-11-05,2014-11-06,Same Day,RH-9495111,Rick Hansen,Consumer,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed",2832.96,8,0.0,311.52,903.04,Critical,Katlego Akosua,NaN


In [59]:
dataset['Descuento'].value_counts()

Descuento
0.000    29009
0.200     4998
0.100     4068
0.400     3177
0.600     2006
0.700     1786
0.500     1633
0.170      735
0.470      725
0.150      541
0.002      461
0.270      388
0.300      340
0.450      327
0.800      316
0.250      198
0.070      150
0.350      122
0.402      104
0.370       74
0.202       41
0.320       27
0.602       23
0.650       17
0.570       12
0.550       10
0.850        2
Name: count, dtype: int64

In [60]:
dataset['Gastos_Envio'].value_counts()

Gastos_Envio
1.360    114
1.350    113
1.790    110
1.970    108
1.800    106
        ... 
1.039      1
1.025      1
1.019      1
1.003      1
1.002      1
Name: count, Length: 16452, dtype: int64

In [61]:
dataset['Gastos_Envio'].describe()

count    51290.000000
mean        26.478567
std         57.251373
min          1.002000
25%          2.610000
50%          7.790000
75%         24.450000
max        933.570000
Name: Gastos_Envio, dtype: float64

Cambiamos la columna devuelto a booleano.

In [62]:
dataset['Devuelto'].value_counts()

Devuelto
Yes    2220
Name: count, dtype: int64

In [63]:
dataset['Devuelto'] = dataset['Devuelto'] == 'Yes'

In [64]:
dataset['Devuelto'].value_counts()

Devuelto
False    49070
True      2220
Name: count, dtype: int64

In [65]:
dataset['Devuelto'].dtype

dtype('bool')

In [66]:
dataset.head()

,ID_Pedido,Fecha_Pedido,Fecha_Envio,Metodo_Envio,ID_Cliente,Nombre_Cliente,Perfil_Cliente,Ciudad,Estado,Pais,Region,Macro_Zona,ID_Producto,Categoria,Subcategoria,Nombre_Producto,Ventas,Cantidad,Descuento,Beneficio,Gastos_Envio,Prioridad_Pedido,Gerente,Devuelto
ID_Venta,,,,,,,,,,,,,,,,,,,,,,,,
40098,CA-2014-AB10015140-41954,2014-11-11,2014-11-13,First Class,AB-100151402,Aaron Bergman,Consumer,Oklahoma City,Oklahoma,United States,Central US,USCA,TEC-PH-5816,Technology,Phones,Samsung Convoy 3,221.98,2,0.0,62.15,40.77,High,Lon Bonher,False
26341,IN-2014-JR162107-41675,2014-02-05,2014-02-07,Second Class,JR-162107,Justin Ritter,Corporate,Wollongong,New South Wales,Australia,Oceania,Asia Pacific,FUR-CH-5379,Furniture,Chairs,"Novimex Executive Leather Armchair, Black",3709.40,9,0.1,-288.77,923.63,Critical,Kauri Anaru,False
25330,IN-2014-CR127307-41929,2014-10-17,2014-10-18,First Class,CR-127307,Craig Reiter,Consumer,Brisbane,Queensland,Australia,Oceania,Asia Pacific,TEC-PH-5356,Technology,Phones,"Nokia Smart Phone, with Caller ID",5175.17,9,0.1,919.97,915.49,Medium,Kauri Anaru,False
13524,ES-2014-KM1637548-41667,2014-01-28,2014-01-30,First Class,KM-1637548,Katherine Murray,Home Office,Berlin,Berlin,Germany,Western Europe,Europe,TEC-PH-5267,Technology,Phones,"Motorola Smart Phone, Cordless",2892.51,5,0.1,-96.54,910.16,Medium,Gilbert Wolff,False
47221,SG-2014-RH9495111-41948,2014-11-05,2014-11-06,Same Day,RH-9495111,Rick Hansen,Consumer,Dakar,Dakar,Senegal,Western Africa,Africa,TEC-CO-6011,Technology,Copiers,"Sharp Wireless Fax, High-Speed",2832.96,8,0.0,311.52,903.04,Critical,Katlego Akosua,False


In [67]:
dataset.info()

<class 'pandas.DataFrame'>
Index: 51290 entries, 40098 to 6147
Data columns (total 24 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   ID_Pedido         51290 non-null  str           
 1   Fecha_Pedido      51290 non-null  datetime64[us]
 2   Fecha_Envio       51290 non-null  datetime64[us]
 3   Metodo_Envio      51290 non-null  str           
 4   ID_Cliente        51290 non-null  str           
 5   Nombre_Cliente    51290 non-null  str           
 6   Perfil_Cliente    51290 non-null  str           
 7   Ciudad            51290 non-null  str           
 8   Estado            51290 non-null  str           
 9   Pais              51290 non-null  str           
 10  Region            51290 non-null  str           
 11  Macro_Zona        51290 non-null  str           
 12  ID_Producto       51290 non-null  str           
 13  Categoria         51290 non-null  str           
 14  Subcategoria      51290 non-null  s

In [ ]:
dataset.to_pickle('dataset_limpio.pkl')

In [69]:
dataset.to_csv('dataset_limpio.csv')

In [70]:
dataset.to_excel('dataset_limpio.xlsx')